pkgs/by-name/ve/veloren/package.nix

```diff
@@ -16,33 +16,30 @@
   libXrandr,
   wayland,
   stdenv,
+  mold,
+  makeWrapper,
 }:
 
 let
   # Note: use this to get the release metadata
-  # https://gitlab.com/api/v4/projects/10174980/repository/tags/v{version}
-  version = "0.17.0";
-  date = "2024-12-28-12:49";
-  rev = "a1be5a7bece7af43ebd76910eb0020c1cf3c0798";
-in
-
-rustPlatform.buildRustPackage {
-  pname = "veloren";
-  inherit version;
+  # https://gitlab.com/api/v4/projects/10174980/repository/tags/nightly
+  version = "0-unstable-2025-11-16";
```

Comment:
No good for me: if you want an unstable release there are the nightly builds distributed via airshipper. This package is supposed to provide the latest stable release.

Reply:
Agreed. But I think it's feasible to have an update script to update it every month or so (since we pin the date, the update script could check it to avoid creating PRs too often).

We could also refactor the code a bit so it's easier to override the attrs for the version, so users can compile it themselves if they really want a specific version.

Reply:
The main problem is that veloren uses unstable rust, so you can't really build a specific version unless the rust toolchain in Nixpkgs (mostly) matches what they use upstream.

What I had done so far was to use the latest release, use stable rust with RUSTC_BOOTSTRAP and backport fixes when the Nixpkgs toolchain is updated and becomes newer than the release, which happens about every 6 months.

I think that if you go with nightly we'll have to deal with far more frequent breackages, which I really don't want to do. Well, unless you become the maintainer...